src/drive/web/modules/services/components/SuggestionProvider.jsx

```diff
@@ -46,44 +47,30 @@ class SuggestionProvider extends React.Component {
     )
   }
 
-  // fetches pretty much all the files and preloads FuzzyPathSearch
+  // fetches pretty much all the files not trashed and preloads FuzzyPathSearch
   async indexFiles() {
     const { client } = this.props
-    // TODO: fix me
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async resolve => {
-      const resp = await cozy.client.fetchJSON(
-        'GET',
-        `/data/io.cozy.files/_all_docs?include_docs=true`
-      )
-      const files = resp.rows
-        // TODO: fix me
-        // eslint-disable-next-line no-prototype-builtins
-        .filter(row => !row.doc.hasOwnProperty('views'))
-        .map(row => ({ id: row.id, ...row.doc }))
 
-      const folders = files.filter(file => file.type === TYPE_DIRECTORY)
+    const files = await client.queryAll(buildSuggestionsQuery().definition)
+
+    const folders = files.filter(file => file.type === TYPE_DIRECTORY)
 
-      const notInTrash = file =>
-        !file.trashed && !/^\/\.cozy_trash/.test(file.path)
-      const notOrphans = file =>
-        folders.find(folder => folder._id === file.dir_id) !== undefined
+    const notInTrash = file =>
+      !file.trashed && !/^\/\.cozy_trash/.test(file.path)
```

Comment:
If you succeed to create the right partialIndex here https://github.com/cozy/cozy-drive/pull/2663/files#r938665394 this map will not be needed

Reply:
changes made in https://github.com/cozy/cozy-drive/pull/2663/commits/59c86d5524666d92c6f711891919cb19813e0adc while @paultranvan is looking at the warning in cozy-client